## Introduction

<!-- motivation; heterogeneity is an obstacle for chemotherapy  -->
One of the primary treatments of cancer consists of chemotherapy, mainly targeted therapies, whereby patients are administered drugs that eliminate fast-proliferating cells to stall cancer growth or eliminate the tumor. Long-term therapeutic efficacy, however, varies significantly due to the vast heterogeneity in intratumor response to therapy [@pmid:16129367; @pmid:20619739]. Cell variability in drug response can originate from cell-intrinsic factors, such as genomic alterations (i.e., altered nucleotide excision repair, telomere maintenance, and copy-number variation) and epigenetic mechanisms like changes in a chromatin state [@pmid:20371346], or cell-extrinsic factors such as spatial variability in the surrounding vasculature and environmental stressors [@pmid:26972587; @pmid:25131830; @pmid:29250983].

<!-- literature review in conventional single-cell variability studies -->
Advances in ‘omics’ technologies have enabled detailed analysis of cell-to-cell variability [@pmid:20619739; @pmid:22397650], and the development of fine mapping and protein network algorithms have determined the presence of causal genetic mutations and dysregulation events that drive abnormal protein function [@pmid:25104515; @pmid:27322546]. These modalities, however, are labor and time-intensive, do not account for environmental factors, and serve primarily as end-point analysis barring longitudinal observation of tumor evolution. In addition to 'omics' modalities, genetic _association_ studies (i.e., Cancer Cell Line Encyclopedia) are similarly able to find common risk factors with smaller effect sizes using population-level samples. The findings are valuable but fail to identify rare and meaningful transitions on the single-cell level [@pmid:22460905], in particular the stochastic changes in individual cell states that have significant effects on overall tumor resistance. Lastly, fitness markers such as cell end-of-life fate, lifetime, and population doubling time are adopted in the clinical setting to measure cell pathologies [@pmid:12640450; @pmid:8072198; @pmid:8655369; @pmid:20981102]. Recent research has made efforts to track phenotypic measurements of fitness at the single-cell level [@pmid:29381859; @pmid:22886092]; however, most efforts are not yet resolved enough to illuminate the full complexity of cancer cells due in large part to reliance on population-level analysis (i.e., IC~50~) [@pmid:25421725].  

<!-- Lineage data is special -->

Measurements accompanied by lineage relationships are uniquely valuable for studying inherited phenotypes within families of people or populations of cells. This value is well-recognized in linkage studies which use pairs of relatives to identify the genetic determinants of disease [@pmid:19136655]. Notably, linkage studies can identify genetic determinants with greater power than even much larger correlation-based studies because relatives essentially serve as internal controls (CITE REVIEW). Linkage studies also start with the phenotype of individuals, rather than grouping based on molecular differences as is most common in cells. Lineage-resolved data has demonstrated unique value in cells for uncovering heterogeneity due to transient differences [@doi:10.1101/373258; @DOI:10.1073/pnas.1715639115].


<!-- tHMMs are a solution to modeling lineage data -->

Hidden Markov models (HMMs) provide a strategy to infer discrete states from measurements when a series of co-dependent observations are made. An example of this is their most widely-used application—time series data—where each measurement is dependent on what came before. Recognizing this co-dependence allows HMMs to make accurate inferences even in the presence of extremely noisy measurements since each neighboring measurement can provide accumulating evidence for a prediction. HMMs have been adapted to lineage trees (tHMMs) so that each measurement across the tree can provide accumulating evidence for a prediction. Just like with time-series data, these models can provide very accurate predictions despite noisy measurements and limited information by recognizing the co-dependence between measurements [@doi:10.1109/78.668544; @doi:10.1109/TSP.2004.832006]. tHMMs have been used in a multitude of applications, from image classification to comparative genomics [@pmid:18255546; @pmid:23762278]. In cells, these models have been fit to lineages collected from stem cells and bacteria colonies [@PMID:19628503; @doi:10.1101/488981]. Improvements in cell tracking and high-throughput imaging promise to make these models valuable techniques for studying the plasticity of heterogeneous cell populations. However, widespread use will require usable implementations that can be readily adapted to different experimental measurements, examples of unique insights they can provide, and standards for experimental validation.

<!-- Introduction to the paper -->

Here, we develop a generic implementation of tHMMs with a defined interface for integrating diverse types of measurements on binary or cell lineage trees. We use this to analyze how populations of single breast cancer cells respond to therapy with a cell cycle reporter. Single cell measurements of the cell cycle revealed extensive variation in drug response that is hidden in population-level measurement. This model allows us to quantify the dynamics and phenotypic features of heterogeneity in drug response. This work, therefore, provides a phenotype-driven route to discovering cell-to-cell variation in drug response, demonstrates an overall strategy for quantifying the dynamics of cell heterogeneity, and implements a very general software tool for widespread use of tHMM models.

