## Introduction

<!-- motivation; heterogeneity is an obstacle for chemotherapy  -->
Chemotherapy and targeted therapies, drugs that selectively eliminate fast-proliferating or oncogene-addicted cells, are among the primary treatments of cancer. Long-term therapeutic efficacy varies significantly due in part to widespread heterogeneity in intratumor response, however [@pmid:16129367; @PMID:20619739]. Cell variability in drug response can originate from cell-intrinsic factors, such as genomic alterations and epigenetic mechanisms like changes in a chromatin state [@pmid:20371346], or cell-extrinsic factors such as spatial variability in the surrounding vasculature and environmental stressors [@doi:10.1038/nrg.2016.13; @pmid:25131830; @pmid:29250983]. Moreover, cell plasticity, where cells adopt the characteristics of other cell types, is observed more often in cancer cells, affecting their sensitivity to therapy [@doi:10.1038/bjc.2015.146].

<!-- literature review in conventional single cell variability studies and that population level fails sometimes-->
Large-scale profiling studies can find molecular features that associate with drug response using population-level samples [@PMID:31068700; @PMID:22460905]. These associations, while valuable, can miss the contribution of cell-to-cell heterogeneity, and especially stochastic changes in individual cell states that have significant effects on overall tumor drug response [@PMID:20371346; @PMID:21854987; @PMID:28607484]. The most common methods for quantifying drug response are simple overall metrics of tumor cell population expansion or contraction [@doi:10.1038/ni908; @pmid:8072198; @pmid:8655369; @pmid:20981102]. Recent research has made efforts to track phenotypic measurements of fitness at the single cell level [@pmid:29381859; @pmid:22886092]; however, even single cell measurements are typically performed with snapshots that subsequently miss the contribution of cells in the overall population response [@pmid:25421725].


<!-- Lineage data is special -->
Measurements accompanied by lineage relationships are uniquely valuable for studying inherited phenotypes within families of people and are likely to be similarly useful in populations of cells. This value is evident in linkage studies wherein relatives are used to identify or refine the genetic determinants of disease [@DOI:10.1126/science.aax3710; @DOI:10.1038/s41467-020-17117-4; @pmid:19136655]. Notably, linkage studies can identify genetic determinants with greater power than even large association studies because relatives essentially serve as internal controls [@PMID:19369670]. Linkage studies also start with the phenotype of individuals, rather than grouping based on molecular differences. Defining heterogeneity phenotypically has previously helped to identify subpopulations that would be hidden if using molecularly-defined clusters [@DOI:10.1016/j.isci.2020.101991]. Recently, constructing phylogenic trees of cancer cells using lineage tracing and single cell sequencing has helped to identify the directionality of metastatic human cancer cells in a mouse xenographt model [@DOI:10.1126/science.abc1944]. Lineage-resolved data has also demonstrated particular value in uncovering cell-to-cell heterogeneity due to transient differences outside of cancer [@doi:10.1101/373258; @doi:10.1073/pnas.1715639115].

<!-- tHMMs are a solution to modeling lineage data -->
Hidden Markov models (HMMs) provide an efficient strategy to infer discrete states from measurements when a series of co-dependent observations are made. An example of this is their wide-spread use in time series analysis, where each measurement is dependent on those that came before [@PMID:15629048; @DOI:10.1109/5.18626]. Recognizing this co-dependence allows HMMs to make accurate inferences even in the presence of extremely noisy measurements since each neighboring measurement provides accumulating evidence for a prediction [@DOI:10.1126/scisignal.aao1917]. HMMs have been adapted to lineage trees (tHMMs) so that each measurement across the tree can similarly provide accumulating evidence for a prediction. Just like with time-series data, these models can provide very accurate predictions despite noisy measurements and limited information by recognizing the co-dependence between related measurements [@doi:10.1109/78.668544; @doi:10.1109/TSP.2004.832006]. tHMMs have been used in a multitude of applications, from image classification to comparative genomics [@pmid:18255546; @pmid:23762278]. In cells, these models have been fit to lineages collected from stem cells and bacteria colonies but have always required tailor-made implementations [@PMID:19628503; @doi:10.1101/488981]. Improvements in cell tracking and high-throughput imaging promise to make these models valuable techniques for studying the plasticity of heterogeneous cell populations. However, widespread use will require usable implementations that can be readily adapted to different experimental measurements, examples of successful tHMM applications that will encourage increased usage, and standards for experimental validation.

<!-- Introduction to the paper -->
Here, we develop a generic implementation of tHMMs with a defined interface for integrating diverse types of measurements on cell lineage trees. We use the relationship between the cells to analyze how populations of breast cancer cells respond to therapy with a cell cycle reporter. Single cell measurements of the cell cycle revealed extensive variation in drug response that is not captured in population-level measurement. This model allows us to quantify the dynamics and phenotypic features of heterogeneity in drug response. We are able to predict the most likely number of phenotypically distinct subpopulations, the characteristics of those subpopulations, transition probabilities from one state to another, and each cell's expected state. This work, therefore, provides a flexible phenotype-driven route to discovering cell-to-cell variation in drug response, demonstrates an overall strategy for quantifying the dynamics of cell heterogeneity, and implements a very general software tool for the widespread use of tHMM models.
