## Results

### Lineage information provides unique information about the source and dynamics of heterogeneity

To illustrate the unique value of lineage measurements when analyzing heterogeneous drug responses, we picked out a series of lineages from our measurements of the breast cancer cell line AU565 treated with gemcitabine (Fig. @fig:lineage). Note that while there is a striking difference in which cells contribute to the population’s growth, the starting and ending numbers of cells are nearly identical. As a result, any measurements of the number of cells on a population level cannot distinguish these two situations. Measurements that record the history of cells (e.g., CFSE staining, Luria-Delbruck experiment) can help to distinguish these two cases but are still limited to finding specific differences [@DOI:10.1073/pnas.1715639115; @PMID:28607484]. Lineage measurements, by contrast, provide sufficiently rich measurements to accurately quantify the extent and dynamics of heterogeneity. By observing individual cells, a variety of phenotypic measurements such as inter-mitotic times, cell cycle phase durations, motility, cell death and division, morphology, and protein and transcription factor levels can be characterized in parallel. On the other hand, it is difficult to disentangle purely stochastic effects among single cells from heritable differences in these data.

![**Population-based measurements are insufficient to distinguish the dynamics of heterogeneity.** Two sets of lineages from AU565 cells treated with gemcitabine.](./output/figure1.svg){#fig:lineage}

### A tHMM infers the state of cells given measurements on lineage trees

Given the unique insights that single-cell measurements on lineage trees can provide, we wished to implement a strategy for predicting the states of cells using these data. Given a set of measurements made across a lineage tree, we use a tree-based hidden Markov model (tHMM) to fit these data (Fig. {@fig:tree}a). Just like a typical hidden Markov model, these models can infer the hidden discrete states of cells given a series of measurements. This takes place using an iterative strategy wherein the states of each cell are predicted given distributions that describe the phenotype of cells in each state (the “expectation” step), and then each distribution of phenotypes is fit to match the cells within that state (the “maximization” step) (Fig. {@fig:tree}b). This expectation-maximization (EM) process repeats until convergence.

After fitting, the tHMM model provides a variety of information about the states (Fig. {@fig:tree}c). First, this process provides an estimate for the starting abundance of each state and the probability of transitions between them. Second, the phenotypes of cells in each state are described and can be compared to distinguish how each state behaves differently. Third, the fit data or new measurements can be used to predict the state of each cell. Finally, the model provides a likelihood of each cell and therefore the data overall. This last quantity can be used, for example, to estimate the number of cell states with distinguishable behavior. Importantly, the phenotype of a cell is only defined by the available measurements and distribution to describe them. When implementing these processes, we ensured this is defined through a modular interface to allow other forms of data to be easily integrated.

![**The tHMM interface.** (a) Input data takes the form of single-cell measurements, where the lineage relationship between cells is known. (b) The fitting process includes expectation and maximization steps, where model parameters are iteratively updated until convergence. (c) Outputs of the model after fitting including the tree of inferred cell states, probabilities of transition between each state, starting abundance of each cell state, and distributions that describe the behavior of cells within each state.](./output/figure2.svg){#fig:tree}

### Completely user-defined states let the user model any population of cells with hertiable phenotypes

<!-- introduction of phenotype choices, walking through how cell fate can be modeled. -->
In our model, cell states are defined by quantifiable differences in cell phenotype such as gene expression, inter-mitotic times, post-translational modification, morphology, and production of proteins [@pmid:30450444]. Additionally, cell states can be grouped by functional properties such as resistance to drugs and tumor-seeding abilities [@pmid:21854987]. This cell heterogeneity, which can be caused by genetic variability as well as non-genetic mechanisms, enables cells to adapt to varying environments and is thus relevant for anticancer drug development [@pmid:20371339]. In our model, we define states as multivariate probabilistic distributions, which in turn, are defined by the set of parameters that describe those distributions. The observed phenotypes for cells in a lineage tree can be viewed as realizations of these multivariate probabilistic distributions.  

<!-- Adding the phase lengths to the observations and explain distributions and parameters associated with it. A touch on censorship. -->

In general, any observable heritable phenotype can be operationalized into a probabilistic distribution that can be used to represent a state, as an instance, using a Gaussian distributed random variable to describe average cell motility as parameterized by its mean and variance. In Fig. @fig:uncenSingle, we validated the model performance by simulating two-state, uncensored single lineages with fate and time observations. We showed that state assignment accuracy approaches 100% and estimation error approaches 0 as cell number increases.

![**This figure shows the flexibility of the model and that we can use any tractable phenotype.** ](./output/figure3.svg){#fig:fate}

### Experiments of finite time necessitate data censorship corrections

Modeling the duration of each cell’s lifetime is complicated by the influence of experimental factors. Specifically, cells at the beginning or end of the experiment are cut off and so, while we observe them, we do not know the duration of their cell cycle phases. One cannot simply remove these cells as this will create a systematic bias in the model; for example, this would create an upper bound on the possible lifetime of a cell. To account for this experimental effect, we can instead observe that we do know a cell lived at least as long as the duration we observe.

To incorporate this effect into our model, we marked cells or their specific cell cycle phase that encountered the bounds of the experiment as censored. When estimating the properties of a state’s lifetime or the probability of a cell’s observation these cells instead used a censored estimator or the survival function of the distribution. Using synthetic data, we were able to verify that we could obtain similar accuracies between lineages that had no cells affected by either cell death or experiment duration and lineages that were affected by these experimental effects (Fig. @fig:censor).

![**Experiments of finite time necessitate data censorship corrections.** (a) Randomly generated uncensored two-state lineages. (b) Randomly generated censored two-state lineages. Uncensored state 0 and 1 cells are shown in different colors. Censored cells are colored by being G1- (pink) or S/G2- (gold) censored. (c) State assignment accuracy with uncensored lineages. (d) State assignment accuracy with experiment time-censored lineages. Scatter points represent the entire fitting process for a randomly generated population.](./output/figure4.svg){#fig:censor}

### Performance on synthetic lineages validates model results

<!-- ("Figure 5") -->

To indicate our model's performance in terms of state assignment accuracy and distribution parameter estimation, we created a synthetic population of cells in uncensore and censored mode for two sets of observations (Fig. @fig:uncenSingle, Fig. @fig:uncenMulti, Fig. @fig:cenMulti, Fig. @fig:performSyn, Fig. @fig:performUncenSingle, Fig. @fig:performUncenMulti, Fig. @fig:performCenMulti). This synthetic data is time and fate-censored to be a valid representative of experimental lineages, however, the true state of individual cells is known as opposed to experimental data. The phenotypic measurements considered in Fig. @fig:performSyn and Fig. @fig:cenMulti include fate and time duration of G1 and S/G2 phase for each cell.

Lineages are created with varying lengths and seeding density (Fig. @fig:performSyn a). Fig. @fig:performSyn depicts the state assignment accuracy, hyper parameter estimation error (transition and initial probability matrices) and parameter estimation errors of Bernoulli (for fate) and gamma distributions (for lifetime). Model reaches roughly $95%$ accuracy for more than 250 cells and the error estimations converge to zero.

![**Model performance on censored lineages of increasing breadth and depth.** (a) Two state populations of increasing breadth (increasing number of lineages or initial cells) and of increasing depth (increasing experiment time) are analyzed. The states are shown in green and blue colors and  red shows cell death. The model performance is shown based on (b) state assignment accuracy, (c) the error bewteen the estimated and the true transition rate matrix and the (d) initial probability vector. In (e-f) it shows the accuracy in estimating the Bernoulli parameters for G1 and S/G2 phase, respectively, and (g) depicts the distance between the ture and estimated gamma distributions associated with phase lengths for the two states.](./output/figure5.svg){#fig:performSyn}


### Identifying states requires partially, but not fully, distinct phenotypes

Cells of even very distinct molecular state can have partly overlapping phenotypes. Therefore, we sought to evaluate how different two states would need to be for us to identify them as distinct (Fig. @fig:wass a). We varied the G1 phase duration from identical to very distinct (Fig. @fig:wass b) and quantified the accuracy of our model. While the phenotypic observation of a given state had to be different for our model to accurately assign cells, even overlapping phenotypes could be distinguished by using the lineage relationships of cells (Fig. @fig:wass c). In this case, the model performance in censored and uncensored populations was relatively identical (Fig. @fig:wass1, Fig. @fig:wass2). This shows that the heritability of a cell state can be used to more accurately identify cells with partially overlapping phenotypes.

![**Model performance relative to the Wasserstein distance between states.** (a) A visual representation of how two states can vary in their phenotypic similarity. (b) The distribution G1 duration is varied in state 1 (blue) while the other state is kept constant. (c) State assignment accuracy versus the Wasserstein distance between state phenotypes.](./output/figure6.svg){#fig:wass}


<!-- ("Figure 7") -->
### The relative presence of states in a population is important because our model depends on non-ancillary sample statistics.

<!-- the importance of recognizing under-represented subpopulations. -->
While the tools we currently have to identify heterogeneity in tumors are limited, this phenomenon indicates that it is necessary to consider underrepresented clusters in solid tumors, such as those that have gained resistance to anticancer drugs through environmental conditions, selective pressure, and cell-state transitions. Thus, we can accurately identify and predict the patterns by which tumors gain drug resistance [@pmid:31449515].

<!-- creating the population that includes under-represented cell states, and explaining the figure. -->
To validate our model's capability to identify under-represented populations, we created a synthetic populations of 2 states and varied the proportion of cells in these states while keeping the total cell number relatively constant. Fig. @fig:prop a represents the successive state proportion changes from the left to the right from 0% to 100%. Fig. @fig:prop b and c depict the accuracy of state assignment corresponding to the proportional changes in Fig. @fig:prop a. It appears that censorship is an important factor when the proportion of cells in different states is not relatively balanced (Fig. @fig:prop4, Fig. @fig:real_5). However, this shows that we can identify under-represented subpopulations with a slightly lower accuracy.

![**Model performance relative to the presence of each state.** (a) A visual representation of how lineages and populations can have different proportions of their total cell population be represented by different states. (b) Populations with different ratios of cells belonging to one state over the other are made. The ratio of the number of cells in state 1 to the number of cells in state 2 increases from 1:10 to 9:10. This is achieved by synthetically creating lineages with transition rates that bias towards making one state over the other. Model performance relative to the presence of that state is shown.](./output/figure7.svg){#fig:prop}

### Model selection effectively identifies the number of distinct states

Usually it is unclear how many distinct cell states may exist within a population. Further, the number of states identified may depend on the environmental context of the cells, particularly because we use phenotypic measurements. To test whether we could determine the number of distinct states, we performed model selection using the Akaike information criterion (AIC) and models with varying numbers of states (Fig. @fig:sAIC). The predicted number of cell states was predominantly correct, and incorrect predictions still centered around the true answer, for both uncensored and censored lineages (Fig. @fig:sAIC). This indicates that model selection can help to identify the appropriate number of cell states given a set of measurements.

![**Model selection effectively identifies the number of distinct states.** (a)-(d) Model AIC for uncensored lineages with 1–4 states. (e)-(h) Model AIC for censored lineages with 1–4 states. AIC values are normalized such that the optimum is equal to 0. Histograms are shown for the minimal AIC over repeated analyses.](./output/figure8.svg){#fig:sAIC}

<!-- figure 9 -->
### The fitting applied on experimental data of phase-specific observations

To show the performance of the model on single-cell experimental data we employed phenotypic measurements of G1 and S/G2 phase lengths and fates of AU565 cells in control condition and drug-treated trials. The experiment was conducted for 96 hours and cells were imaged every 30 minutes and tracked over time. Gemcitabine and lapatinib treatment data with lineages more than 5 cells were imported and fitted to the model. We calculated the AIC metric for several concentrations of each drug, separately. The AIC values for all conditions of a drug were accumulated assuming the influence of a specific drug treatment in terms of the number of pheotypic states of its target population is not concentration-dependent. Fig. @fig:expAIC depicts the most likely number of states that the model finds using AIC metric for lapatinib (Fig. @fig:expAIC a) and gemcitabine (Fig. @fig:expAIC b). 


![**AIC for experimental data among 4 states of phase lengths.** (a)-(d) AIC curves for control condition and 3 concentrations of lapatinib. (e)-(h) AIC curves for control condition and 3 concentrations of gemcitabine.](./output/figure9.svg){#fig:expAIC}

### Estimated emissions explain the influence of drug treatment

Explain me.

![**Estimated Emissions of the experimental data** ](./output/figure11.svg){#fig:emissions}
