## Results

<!-- # "Figure 1" -->
### The tHMM model can disentangle phenotypic heterogeneity in a single-cell tracked population.

As a motivating trivial scenario, we consider "tumor growth rate" as the phenotype of interest operationalized by a cell's lifetime and its fate in a lineage and present two cases of cell population growth over time. The cases include one where the population is homogeneous and all cells show consistent behaviour across their observed phenotype (Fig. @fig:lineage a) and the other where the population can be clustered into at least two subpopulations, each with different growth rates shown in different colors (Fig. @fig:lineage b). Over time, the two exponentially growing populations look similar with regards to their individual lineages, their seeding density (how they are initialized) and their final cell count at end of the experiment. Thus, the estimated collective tumor growth rate calculated via existing population-level analysis for both cases will inevitably be equal, and the heterogeneity existing in the second case will be masked. Utilizing the tHMM model we have built, we are not only able to capture the heterogeneity present in the population shown in Fig. @fig:lineage b, but also able to quantify the relative presence of these two states, summarizing the characteristics of the observed subpopulations as probabilstic distributions. Although the growth rate (or doubling time) yielded by the usual population-level methods and the tHMM for the homogeneous case are equivalent, without recognizing the heterogeneity in the population, usual methods fail in predicting the long-term growth properties in the second case correctly.

The capability to identify phenotypic heterogeneity in a population of cancer cells is critical when designing anti-cancer regimens and drug combination protocols [@doi:10.1016/j.semcancer.2016.11.006; @doi:10.1126/scisignal.aas8779]. Cell kinetics such as inter-mitotic times, cell cycle phase durations, death and division, and other phenotypic characteristics such as cell morphology, protein and transcription factor levels, etc., are important targets for cancer chemotherapy, hence, important to identify subpopulations with distinct phenotypic responses after drug exposures. Given single-cell data of any heritable phenotype, our proposed model can disentangle subpopulations that respond to the exposure differently from one another.  

![**Identifying heterogeneity in cell population.** (a) A homogeneous population of cells reaching 18 cells at the end of the experiment. (b) A heterogeneous population of cells consisting of 2 states reaching 18 cells.](./output/figure1.svg){#fig:lineage}

<!-- # (Figure 2) -->
### Model interface including inputs and outputs  

In this section, the interface of the model and possible input/outputs are explained. The input raw data is assumed to be a quantitative measurements of some phenotype in single-cell level, such as inter-mitotic times, cell cycle phase duration, etc., in which the relationship between the cells is preserved as a binary tree, shown in Fig. @fig:tree a. The tHMM takes in lineages in the form of an object in Python which consists of a list of cells within that lineage. Each individual cell is an object with its own properties such as observations, state (which is assigned to it after fitting), its start and end time (relative to the start of the experiment), generation, etc., and accessibility to its ancestors and decendants. The proposed package includes instructions and functions to create objects of cells and lineages from an input data file. After creating the lineage objects, the data is inputted to the tHMM for fitting where an expectation maximization (EM) algorithm runs until convergence (Fig. @fig:tree b). The Expectation step takes in an estimation of the model's parameters and the tree of the observations, and outputs the most likely prediction of the states tree. In the maximization step, a prediction of states tree and the observations tree are inputted, and the most likely estimation of model parameters associated with observations are outputted.

The tHMM can provide various statistical outputs as a result of fitting, such as the probability of transition between the states, the most likely state of each individual cell, the most fitted distribution associated with each state, and the most likely number of states in a population (Fig. @fig:tree c). To address the possibility of incorporating numerous phenotypes that can cause heterogeneity within a population, the model is made flexible in terms of the types of heritable measurements that could be used as clustering features. As shown in Fig. @fig:tree a, any observation can be added into the property list of individual cells. Any new phenotype requires the user to define a possible distribution to which the phenotype belongs.
  

![**The interface of tHMM.** (a) The input data gets converted to the lineage objects with cells instantiated with properties from the data, such as cells' observations, parent-daughter link, generation number, etc. (b) The fitting process includes expectation and maximization steps. (c) Outputs of the model after fitting including the tree of states, the tree of observations, the transition probabilities, and the most likely distribution parameters.](./output/figure2.svg){#fig:tree}

<!--  ("Figure 3") -->
### Completely user-defined states let the user model any population of cells with hertiable phenotypes

<!-- introduction of phenotype choices, walking through how cell fate can be modeled. -->
Because the goal of our model is to identify cell states, we have packaged our model with a lightweight framework for developing and using states. Cell states are defined by quantifiable differences in cell phenotype such as gene expression, inter-mitotic times, post-translational modification, morphology, and production of proteins [@pmid:30450444]. Additionally, cell states can be grouped by functional properties such as resistance to drugs and tumor-seeding abilities [@pmid:21854987]. This cell heterogeneity, which can be caused by genetic variability as well as non-genetic mechanisms, enables cells to adapt to varying environments and is thus relevant for anticancer drug development [@pmid:20371339]. In our model, we define states as multivariate probabilistic distributions, which in turn, are defined by the set of parameters that describe those distributions. The observed phenotypes for cells in a lineage tree can be viewed as realizations of these multivariate probabilistic distributions. Ultimately, the goal of our model is to identify an optimal number of states, identify which cells belong to which states, and have the states represent the observed phenotypes of the cells in that state. 

<!-- Adding the phase lengths to the observations and explain distributions and parameters associated with it. A touch on censorship. -->
In the case that we present, we observe the times that each cell spent in both G1 and S/G2, as well as the fate observed at the end of each of those phases. In general, any observable heritable phenotype can be operationalized into a probabilistic distribution that can be used to represent a state, such as using a Gaussian distributed random variable to describe average cell motility as parameterized by its mean and variance. In Fig. @fig:uncenSingle, we validated the model performance by simulating two-state, uncensored single lineages with fate and time observations. We showed that state assignment accuracy approaches 100% and estimation error approaches 0 as cell number increases.

![**This figure shows the flexibility of the model and that we can use any tractable phenotype.** ](./output/figure3.svg){#fig:fate}

### Experiments of finite time necessitate data censorship corrections

Modeling the duration of each cell’s lifetime is complicated by the influence of experimental factors. Specifically, cells at the beginning or end of the experiment are cut off and so, while we observe them, we do not know the duration of their cell cycle phases. One cannot simply remove these cells as this will create a systematic bias in the model; for example, this would create an upper bound on the possible lifetime of a cell. To account for this experimental effect, we can instead observe that we do know a cell lived at least as long as the duration we observe.

To incorporate this effect into our model, we marked cells or their specific cell cycle phase that encountered the bounds of the experiment as censored. When estimating the properties of a state’s lifetime or the probability of a cell’s observation these cells instead used a censored estimator or the survival function of the distribution. Using synthetic data, we were able to verify that we could obtain similar accuracies between lineages that had no cells affected by either cell death or experiment duration and lineages that were affected by these experimental effects (Fig. @fig:censor).

![**Experiments of finite time necessitate data censorship corrections.** (a) Randomly generated uncensored two-state lineages. (b) Randomly generated censored two-state lineages. Uncensored state 0 and 1 cells are shown in different colors. Censored cells are colored by being G1- (pink) or S/G2- (gold) censored. (c) State assignment accuracy with uncensored lineages. (d) State assignment accuracy with experiment time-censored lineages. Scatter points represent the entire fitting process for a randomly generated population.](./output/figure4.svg){#fig:censor}

### Performance on synthetic lineages validates model results

<!-- ("Figure 5") -->
To indicate our model's performance in terms of state assignment accuracy and distribution parameter estimation, we created a synthetic population of cells in uncensore and censored mode for two sets of observations (Fig. @fig:uncenSingle, Fig. @fig:uncenMulti, Fig. @fig:cenMulti, Fig. @fig:performSyn, Fig. @fig:performUncenSingle, Fig. @fig:performUncenMulti, Fig. @fig:performCenMulti). We created censored synthetic data such that lineages possess the properties of real experimental data where the true state of individual cells is known. The phenotypic measurements considered in Fig. @fig:performSyn and Fig. @fig:cenMulti include: 1- whether a cell survives G1 phase and transitions to S/G2 phase, 2- whether a cell survives G2 phase and divides, 3- G1 phase duration, and 4- S/G2 phase duration. The fate and duration phenotypes are modeled by a Bernoulli and a gamma distribution, respectively. For a population with 2 known states, we need two sets of different parameters corresponding to the distributions.

Lineages are created with varying lengths and seeding density (Fig. @fig:performSyn a) with the state assignment accuracy that is represented in Fig. @fig:performSyn b. It appears that $~150$ cells are enough to reach $95%$ accuracy. Fig. @fig:performSyn c and d show the difference between the predicted and true transition matrix and initial probability matrix and are decreasing as the population grows larger. Fig. @fig:performSyn e and f depict the estimated fate parameters versus the true parameters, and Fig. @fig:performSyn g represents the distance between the true and estimated gamma distributions of the two states decrease for larger cell numbers. Evidently, the model performs fairly accuracte for lineages with $~150$ or more cells.

![**Model performance on censored lineages of increasing breadth and depth.** (a) Two state populations of increasing breadth (increasing number of lineages or initial cells) and of increasing depth (increasing experiment time) are analyzed. The model performance as based on (b) state assignment accuracy, (c) the error bewteen the estimated and the true transition rate matrix and the (d) initial probability vector, (e-f) the accuracy in estimating the Bernoulli parameters for G1 and S/G2 phase, respectively, and (g) the distance between the ture and estimated gamma distributions associated with phase lengths for the two states are shown.](./output/figure5.svg){#fig:performSyn}


### Identifying states requires partially, but not fully, distinct phenotypes

Cells of even very distinct molecular state can have partly overlapping phenotypes. Therefore, we sought to evaluate how different two states would need to be for us to identify them as distinct (Fig. @fig:wass a). We varied the G1 phase duration from identical to very distinct (Fig. @fig:wass b) and quantified the accuracy of our model. While the phenotypic observation of a given state had to be different for our model to accurately assign cells, even overlapping phenotypes could be distinguished by using the lineage relationships of cells (Fig. @fig:wass c). In this case, the model performance in censored and uncensored populations was relatively identical (Fig. @fig:wass1, Fig. @fig:wass2). This shows that the heritability of a cell state can be used to more accurately identify cells with partially overlapping phenotypes.

![**Model performance relative to the Wasserstein distance between states.** (a) A visual representation of how two states can vary in their phenotypic similarity. (b) The distribution G1 duration is varied in state 1 (blue) while the other state is kept constant. (c) State assignment accuracy versus the Wasserstein distance between state phenotypes.](./output/figure6.svg){#fig:wass}


<!-- ("Figure 7") -->
### The relative presence of states in a population is important because our model depends on non-ancillary sample statistics.

<!-- the importance of recognizing under-represented subpopulations. -->
While the tools we currently have to identify heterogeneity in tumors are limited, this phenomenon indicates that it is necessary to consider underrepresented clusters in solid tumors, such as those that have gained resistance to anticancer drugs through environmental conditions, selective pressure, and cell-state transitions. Thus, we can accurately identify and predict the patterns by which tumors gain drug resistance [@pmid:31449515].

<!-- creating the population that includes under-represented cell states, and explaining the figure. -->
To validate our model's capability to identify under-represented populations, we created a synthetic populations of 2 states and varied the proportion of cells in these states while keeping the total cell number relatively constant. Fig. @fig:prop a represents the successive state proportion changes from the left to the right from 0% to 100%. Fig. @fig:prop b and c depict the accuracy of state assignment corresponding to the proportional changes in Fig. @fig:prop a. It appears that censorship is an important factor when the proportion of cells in different states is not relatively balanced (Fig. @fig:prop4, Fig. @fig:real_5). However, this shows that we can identify under-represented subpopulations with a slightly lower accuracy.

![**Model performance relative to the presence of each state.** (a) A visual representation of how lineages and populations can have different proportions of their total cell population be represented by different states. (b) Populations with different ratios of cells belonging to one state over the other are made. The ratio of the number of cells in state 1 to the number of cells in state 2 increases from 1:10 to 9:10. This is achieved by synthetically creating lineages with transition rates that bias towards making one state over the other. Model performance relative to the presence of that state is shown.](./output/figure7.svg){#fig:prop}

### Model selection effectively identifies the number of distinct states

Usually it is unclear how many distinct cell states may exist within a population. Further, the number of states identified may depend on the environmental context of the cells, particularly because we use phenotypic measurements. To test whether we could determine the number of distinct states, we performed model selection using the Akaike information criterion (AIC) and models with varying numbers of states (Fig. @fig:sAIC). The predicted number of cell states was predominantly correct, and incorrect predictions still centered around the true answer, for both uncensored and censored lineages (Fig. @fig:sAIC). This indicates that model selection can help to identify the appropriate number of cell states given a set of measurements.

![**Model selection effectively identifies the number of distinct states.** (a)-(d) Model AIC for uncensored lineages with 1–4 states. (e)-(h) Model AIC for censored lineages with 1–4 states. AIC values are normalized such that the optimum is equal to 0. Histograms are shown for the minimal AIC over repeated analyses.](./output/figure8.svg){#fig:sAIC}

<!-- figure 9 -->
### The fitting applied on experimental data of phase-specific observations

To show the performance of the model on single-cell experimental data we employed phenotypic measurements of G1 and S/G2 phase lengths and fates of AU565 cells in control condition and drug-treated trials. The experiment was conducted for 96 hours and cells were imaged every 30 minutes and tracked over time. Gemcitabine and lapatinib treatment data with lineages more than 5 cells were imported and fitted to the model. We calculated the AIC metric for several concentrations of each drug, separately. The AIC values for all conditions of a drug were accumulated assuming the influence of a specific drug treatment in terms of the number of pheotypic states of its target population is not concentration-dependent. Fig. @fig:expAIC depicts the most likely number of states that the model finds using AIC metric for lapatinib (Fig. @fig:expAIC a) and gemcitabine (Fig. @fig:expAIC b). 


![**AIC for experimental data among 4 states of phase lengths.** (a)-(d) AIC curves for control condition and 3 concentrations of lapatinib. (e)-(h) AIC curves for control condition and 3 concentrations of gemcitabine.](./output/figure9.svg){#fig:expAIC}
