## Results

### Lineage information provides unique information about the source and dynamics of heterogeneity

To illustrate the unique value of lineage measurements when analyzing heterogeneous drug responses, we picked out a series of lineages from our measurements of the breast cancer cell line AU565 treated with gemcitabine (Fig. @fig:lineage). Note that while there is a striking difference in which cells contribute to the population’s growth, the starting and ending numbers of cells are nearly identical. As a result, any measurements of the number of cells on a population level cannot distinguish these two situations. Measurements that record the history of cells (e.g., CFSE staining, Luria-Delbruck experiment) can help to distinguish these two cases but are still limited to finding specific differences [@DOI:10.1073/pnas.1715639115; @PMID:28607484]. Lineage measurements, by contrast, provide sufficiently rich measurements to accurately quantify the extent and dynamics of heterogeneity. By observing individual cells, a variety of phenotypic measurements such as inter-mitotic times, cell cycle phase durations, motility, cell death and division, morphology, and protein and transcription factor levels can be characterized in parallel. On the other hand, it is difficult to disentangle purely stochastic effects among single cells from heritable differences in these data.

![**Population-based measurements are insufficient to distinguish the dynamics of heterogeneity.** Two sets of lineages from AU565 cells treated with gemcitabine.](./output/figure1.svg){#fig:lineage}

### A tHMM infers the state of cells given measurements on lineage trees

Given the unique insights that single-cell measurements on lineage trees can provide, we wished to implement a strategy for predicting the states of cells using these data. Given a set of measurements made across a lineage tree, we use a tree-based hidden Markov model (tHMM) to fit these data (Fig. {@fig:tree}a). Just like a typical hidden Markov model, these models can infer the hidden discrete states of cells given a series of measurements. This takes place using an iterative strategy wherein the states of each cell are predicted given distributions that describe the phenotype of cells in each state (the “expectation” step), and then each distribution of phenotypes is fit to match the cells within that state (the “maximization” step) (Fig. {@fig:tree}b). This expectation-maximization (EM) process repeats until convergence.

After fitting, the tHMM model provides a variety of information about the states (Fig. {@fig:tree}c). First, this process provides an estimate for the starting abundance of each state and the probability of transitions between them. Second, the phenotypes of cells in each state are described and can be compared to distinguish how each state behaves differently. Third, the fit data or new measurements can be used to predict the state of each cell. Finally, the model provides a likelihood of each cell and therefore the data overall. This last quantity can be used, for example, to estimate the number of cell states with distinguishable behavior. Importantly, the phenotype of a cell is only defined by the available measurements and distribution to describe them. When implementing these processes, we ensured this is defined through a modular interface to allow other forms of data to be easily integrated.

![**The tHMM interface.** (a) Input data takes the form of single-cell measurements, where the lineage relationship between cells is known. (b) The fitting process includes expectation and maximization steps, where model parameters are iteratively updated until convergence. (c) Outputs of the model after fitting including the tree of inferred cell states, probabilities of transition between each state, starting abundance of each cell state, and distributions that describe the behavior of cells within each state.](./output/figure2.svg){#fig:tree}

### Completely user-defined states let the user model any population of cells with hertiable phenotypes

<!-- introduction of phenotype choices, walking through how cell fate can be modeled. -->
In our model, cell states are defined by quantifiable differences in cell phenotype such as gene expression, inter-mitotic times, post-translational modification, morphology, and production of proteins [@pmid:30450444]. Additionally, cell states can be grouped by functional properties such as resistance to drugs and tumor-seeding abilities [@pmid:21854987]. This cell heterogeneity, which can be caused by genetic variability as well as non-genetic mechanisms, enables cells to adapt to varying environments and is thus relevant for anticancer drug development [@pmid:20371339]. In our model, we define states as multivariate probabilistic distributions, which in turn, are defined by the set of parameters that describe those distributions. The observed phenotypes for cells in a lineage tree can be viewed as realizations of these multivariate probabilistic distributions.  

<!-- Adding the phase lengths to the observations and explain distributions and parameters associated with it. A touch on censorship. -->

In general, any observable heritable phenotype can be operationalized into a probabilistic distribution that can be used to represent a state, as an instance, using a Gaussian distributed random variable to describe average cell motility as parameterized by its mean and variance. In Fig. @fig:uncenSingle, we validated the model performance by simulating two-state, uncensored single lineages with fate and time observations. We showed that state assignment accuracy approaches 100% and estimation error approaches 0 as cell number increases.

![**This figure shows the flexibility of the model and that we can use any tractable phenotype.** ](./output/figure3.svg){#fig:fate}

### Experiments of finite time necessitate data censorship corrections

Modeling the duration of each cell’s lifetime is complicated by the influence of experimental factors. Specifically, cells at the beginning or end of the experiment are cut off and so, while we observe them, we do not know the duration of their cell cycle phases. Previously, this challenged has been addressed by simply removing the unfinished cells [@doi:10.7554/eLife.51002]. However, doing so especially in small-size experiments may result in losing a great portion of the data. On the other hand, taking these truncated values into account as a fully observed sample, may create a systematic bias in the model; for example, this would create an upper bound on the possible lifetime of a cell. To account for this experimental effect, we can instead observe that we do know a cell lived at least as long as the duration we observe.

To incorporate this effect into our model, we marked cells or their specific cell cycle phase that encountered the bounds of the experiment as censored. When estimating the properties of a state’s lifetime or the probability of a cell’s observation these cells instead used a censored estimator or the survival function of the distribution. Using synthetic data, we verified that this correction provided similar cell state prediction accuracies between lineages with no cells affected by either cell death or experiment duration and lineages that were affected by these experimental effects (Fig. @fig:censor). This ensures that we can accurately infer cell states despite these experimental limitations.

![**Experiments of finite time necessitate data censorship corrections.** (a) Randomly generated uncensored two-state lineages. (b) Randomly generated censored two-state lineages. State 0 and 1 cells are shown in different colors. (c) State assignment accuracy with uncensored lineages. (d) State assignment accuracy with experiment time- and cell death-censored lineages. Scatter points represent the entire fitting process for a randomly generated population.](./output/figure4.svg){#fig:censor}

### Synthetic lineage benchmarks show a tHMM can accurately estimate population behavior

To evaluate how accurately a tHMM model could infer the behavior of multi-state cell populations, we used synthetic populations of cells in a wide variety of configurations. In each case we determined that our implementation of a tHMM model could accurately and without bias infer the hidden states and parameters of a population given roughly 100 cells. This synthetic data included situations that were not censored (Figs. @fig:uncenSingle, @fig:uncenMulti, @fig:performUncenSingle, @fig:performUncenMulti) and censored due to cell death and experimental duration to be a valid representative of experimental linea g:performSyn, @fig:performCenMulti, @fig:cenMulti). In addition to varying the number of cells in a population, we benchmarked populations with varied percentages of cell states (Figs. @fig:prop, @fig:prop4, @fig:real_5) and varied phenotypic differences between the states (Figs. @fig:wass1, @fig:wass2, @fig:wass). In total, this benchmarking showed that our tHMM model would provide accurate results.

More specifically, one of the benchmarking studies we performed was with data matching our measurements of AU565, where G1 and S/G2 phase periods were quantified (Fig. {@fig:performSyn}a). Although the tHMM model was fit with no information about the true underlying parameters of the simulated cells, it accurately distinguished the two underlying cell states phenotypes (Fig. {@fig:performSyn}b–d) and member cells (Fig. {@fig:performSyn}e). On the population level, the transition and starting probabilities were accurately estimated (Fig. {@fig:performSyn}f–g). Thus, we are confident that, with similar experimental data, we should derive accurate results.

![**Model performance on censored lineages of increasing breadth and depth.** (a) Two state populations of increasing breadth (increasing number of lineages or initial cells) and of increasing depth (increasing experiment time) are analyzed. The states are shown in green and blue colors and red shows cell death. The model performance is shown based on (b) state assignment accuracy, (c) the error between the estimated and the true transition rate matrix and the (d) initial probability vector. In (e-f) it shows the accuracy in estimating the Bernoulli parameters for G1 and S/G2 phase, respectively, and (g) depicts the distance between the true and estimated Gamma distributions associated with phase lengths for the two states.](./output/figure5.svg){#fig:performSyn}


### Identifying states requires partially, but not fully, distinct phenotypes

Cells of even very distinct molecular states can have partly overlapping phenotypes. Therefore, we sought to evaluate how different two states would need to be for us to accurately identify them as distinct (Fig. {@fig:wass}a). We varied the G1 phase duration from identical to very distinct (Fig. {@fig:wass}b) and quantified the accuracy of our model. While the phenotypic observation of a given state had to be different for our model to accurately assign cells, even partly overlapping phenotypes could be distinguished by using the lineage relationships of cells (Fig. {@fig:wass}c). Importantly, our model consistently out-performed clustering the phenotypic measurements without lineage relationships (Fig. {@fig:wass}c). The model performance in censored and uncensored populations were relatively identical (Fig. @fig:wass1, @fig:wass2). This shows that the heritability of a cell state can be used to more accurately identify cells with partially overlapping phenotypes.

![**Model performance relative to the Wasserstein distance between states.** (a) A visual representation of how two states can vary in their phenotypic similarity. (b) The distribution G1 duration is varied in state 1 (blue) while the other state is kept constant. (c) State assignment accuracy versus the Wasserstein distance between state phenotypes.](./output/figure6.svg){#fig:wass}

### Rare cell states require greater numbers of cells

To validate our model's capability to identify less common cell states, we created a synthetic populations of 2 states and varied the proportion of cells in these states while keeping the total cell number relatively constant (Fig. {@fig:prop}a). Not surprisingly, the cell state assignment accuracy decreased when a cell state became rare within the population (Fig. {@fig:prop}b–c). Importantly, we used a balanced accuracy metric when quantifying the accuracy of cell state assignment. This means that rare cell states are weighted more heavily in this accuracy calculation. We observed that most incorrect assignments were of the less common cell state. Therefore, these results indicated that one must have multiple measurements of a given cell state to accurately quantify its presence and behavior. However, more abundant states can still be accurately quantified in the presence of rare states with insufficient cells (Fig. @fig:prop4, @fig:real_5).

![**Model performance relative to the presence of each state.** (a) A visual representation of how lineages and populations can have different proportions of their total cell population be represented by different states. (b) Cell state assignment accuracy for benchmark synthetic cell populations with varied ratios of each cell state. (c) Cell state assignment accuracy for censored synthetic cell populations with varied ratios of each cell state.](./output/figure7.svg){#fig:prop}

### Likelihood-based model selection can effectively identify the number of distinct states

Usually it is unclear how many distinct cell states may exist within a population. Further, the number of states identified may depend on the environmental context of the cells, particularly because we use phenotypic measurements. To test whether we could determine the number of distinct states, we performed model selection using the Akaike information criterion (AIC) and models with varying numbers of states (Fig. @fig:sAIC). The predicted number of cell states was predominantly correct, and incorrect predictions still centered around the true answer, for both uncensored and censored lineages (Fig. @fig:sAIC). This indicates that model selection can help to identify the appropriate number of cell states given a set of measurements.

![**Model selection effectively identifies the number of distinct states.** (a)-(d) Model AIC for uncensored lineages with 1–4 states. (e)-(h) Model AIC for censored lineages with 1–4 states. AIC values are normalized such that the optimum is equal to 0. Histograms are shown for the minimal AIC over repeated analyses.](./output/figure8.svg){#fig:sAIC}

<!-- figure 9 -->
### The fitting applied on experimental data of phase-specific observations

To show the performance of the model on single-cell experimental data we employed phenotypic measurements of G1 and S/G2 phase lengths and fates of AU565 cells in control condition and drug-treated trials. The experiment was conducted for 96 hours and cells were imaged every 30 minutes and tracked over time. Gemcitabine and lapatinib treatment data with lineages more than 5 cells were imported and fitted to the model. We calculated the AIC metric for several concentrations of each drug, separately. The AIC values for all conditions of a drug were accumulated assuming the influence of a specific drug treatment in terms of the number of phenotypic states of its target population is not concentration-dependent. Fig. @fig:expAIC depicts the most likely number of states that the model finds using AIC metric for lapatinib (Fig. {@fig:expAIC}a) and gemcitabine (Fig. {@fig:expAIC}b). 


![**AIC for experimental data among 4 states of phase lengths.** (a) AIC curves for control condition and 3 concentrations of lapatinib. (b) AIC curves for control condition and 3 concentrations of gemcitabine.](./output/figure9.svg){#fig:expAIC}

### Estimated emissions explain the influence of drug treatment

To visualize the properties of the states that the model has identified, we plotted the distribution of average phase-specific life time which in this case is calculated by the multiplication of the two Gamma parameters.  Figs. {@fig:emissions}a-d show that treatment with lapatinib leads to extension of G1 phase in state 0, and gemcitabine treatment shown in Figs. {@fig:emissions}e-h) shows G2 phase extension as a result of increasing the concentration. Both have been experimentally observed.

![**Estimated Emissions of the experimental data** ](./output/figure11.svg){#fig:emissions}
