## Results

### Our model provides more insight than population-level analyses into data with heterogeneous cell-types.

<!-- # "Figure 1" -->

As a motivating trivial scenario, we present two cases of cell population growth over time, one where the population is homogeneous and all cells show consistent behaviour across their observed phenotype and one where the population can be clustered into at least two subpopulations, each with different growth rates (Figure 1b). To further our motivation, we consider "tumor growth rate" as the phenotype of interest operationalized by a cell's lifetime and its fate in a lineage. Identification of tumor growth rate has a wide range of applications in prognostics and can provide quantified therapeutic effects while different treatment modalities are in use [@doi:10.1111/j.1349-7006.1990.tb02591.x; @doi:10.7326/0003-4819-89-1-107]. Differing tumor growth rates can imply the presence of multiple subpopulations of tumors in a cancer; such subpopulations can show variablility in drug response as well. Figure 1a shows the usual view of cell lineages, where a homogeneous population contains cells that are each dividing in a similar fashion; Figure 1b shows a heterogeneous population of cells clustered into two distinct states shown in two different colors. Over time, the two populations look similar with regards to their individual lineages, their seeding density (how they are initialized) and their final cell count at end of the experiment. Assuming that both popluations undergo exponential growth, the doubling rate, and consequently, the estimated collective tumor growth rate, for both cases will inevitably be equal. Via these usual methods of analysis, the heterogeneity of the second population has been masked out and not captured. Ideally, a method used to analyze the growth properties of lineage data should be able to identify the distinct subpopulations within the lineages, and take their presence into account when calculating population-level insights. Utilizing the tHMM model we have built, we are not only able to capture the heterogeneity present in the population shown in Figure 1b, but also able to quantify the relative presence of these two states, summarizing the characteristics of the observed subpopulations as probabilstic distributions. Although the growth rate (or doubling time) yielded by the usual population-level methods and the tHMM for the homogeneous case are equivalent, without recognizing the heterogeneity in the population, usual methods fail in predicting the long-term growth properties in the second case correctly.

As mentioned before, the capability to identify phenotypic heterogeneity in a population of cancer cells is critical when designing anti-cancer regimens and drug combination protocols. Cell kinetics such as inter-mitotic times, cell cycle phase durations, death and division, and other phenotypic characteristics such as cell morphology, protein and transcription factor levels, etc., are important targets for cancer chemotherapy, hence, important to identify subpopulations with distinct phenotypic responses after drug exposures. Given single-cell data of any heritable phenotype, our proposed model can disentgle subpopulations that respond to the exposure differently from one another.  

![**Identifying heterogeneity in cell population.** (a) A homogeneous population of cells reaching 18 cells at the end of the experiment. (b) A heterogeneous population of cells consisting of 2 states reaching 18 cells.](./output/figure1.svg){#fig:1}

### Model interface including inputs and outputs  

<!-- # (Figure 2) -->

In this section, the interface of the model and possible input/outputs are explained. The input raw data is assumed to be a quantitative measurements of some phenotype in single-cell level, such as inter-mitotic times, cell cycle phase duration, etc., in which the relationship between the cells is preseved as a binary tree, shown in Figure 2a. The tHMM takes in lineages in the form of an object in Python which consists of a list of cells within that lineage. Each individual cell is an object with its own properties associated with it such as cell's observations, cell's state (which is assigned to it after fitting), its start and end time (relative to the start of the experiment), cell's generation, etc., and accessibility to its ancestors and decendants. The proposed package includes instructions and functions to create objects of cells and lineages from an input data file. After creating the lineage objects, a list of lineages are inputted to the tHMM for fitting where an expectation maximization (EM) algorithm runs until convergence (Figure 2b). The Expectation step takes in an estimation of the model's parameters and the tree of the observations, and outputs the most likely prediction of the tree of states. In the maximization step, a prediction of tree of states and the tree of observations are inputted, and the most likely estimation of model parameters associated with observations are outputted.

The tHMM can provide various statistical outputs as a result of fitting, such as the probability of transition between the states, the most likely state of each individual cell, the most fitted distribution associated with each state, and the most likely number of states in a population (Figure 2c). To address the possibility of incorporating numerous phenotypes that can cause heterogeneity within a population, the model is made flexible in terms of the types of heritable measurements that could be used as clustering features. As shown in figure 2a, any observation can be added into the property list of individual cells. Any new phenotype requires the user to define a possible distribution to which the phenotype belongs.
  

![**The interface of tHMM.** (a) The input data gets converted to the lineage objects with cells instantiated with properties from the data, such as cells' observations, paret-daughter link, generation number, etc. (b) The fitting process includes expectation and maximization steps. (c) Outputs of the model after fitting including the tree of states, the tree of observations, the transition probabilities, and the most likely distribution parameters.](./output/figure2.svg){#fig:2}

<!--  ("Figure 3") -->
### States are completely user-defined and virtually any population of cells with hertiable phenotypes can be modeled.

<!-- introduction of phenotype choices, walking through how cell fate can be modeled. -->
Because the goal of our model is to identify states, we have packaged with our model a light-weight framework for developing and using states. Cell states are defined by quantifiable differences in cell phenotype such as gene expression, inter-mitotic times, post-translational modification, morphology, and production of proteins [@pmid:30450444]. Additionally, cell states can be grouped by functional properties such as resistance to drugs and tumor-seeding abilities [@pmid:21854987]. Obtaining quantifiable data of cell state transitions across a lineage has relevant applications in bioengineering and drug development such as understanding and predicting the dynamic nature of cancer cells acquiring drug resistance. In systems biology, populations of cells are grouped by cell state and the data collected can be used to predict an accurate variable response in cell behavior. This cell heterogeneity, which can be caused by genetic variability as well as non-genetic mechanisms, enables cells to adapt to varying environments and is thus relevant for the study of anticancer drug development [@pmid:20371339]. We define states as multivariate probabilistic distributions, which in turn, are defined by the set of parameters that describe those very distributions. The observed phenotypes for cells in a lineage tree can be viewed as realizations of these multivariate probabilistic distributions. As a simple example, we can consider only observing the end fate of a cell. Interpreting this as a random variable, the fate of a cell can be represented as the outcome of a Bernoulli random trial: if a cell lives and divides, then the random variable is 1 and if a cell dies, then the random variable is 0. For each cell, the observed phenotype of its fate, either 1 or 0, can be seen as a sampling of the state it belongs to. The Bernoulli probability distribution is parameterized by a single rate parameter which describes the rate at which 1 is more likely than 0 as the outcome of a random trial. In this simple scenario, each state is another Bernoulli probability distribution, defined by its unique rate parameter. Ultimately, the goal of our model is to identify an optimal number of states, identify which cells belong to which states, and have the states represent the observed phenotypes of the cells in that state. Continuing the same example, if the model were to look for two states in a given cell population, then it might define the cells that divide more often into the first state, and the other set of cells that die more often into a second state. Because each state in this example is a Bernoulli probability distribution that describes the observed phenotypes of the cells that belong to that state, the first state will have a Bernoulli parameter that is larger than that of the second state. 

<!-- Adding the phase lengths to the observations and explain distributions and parameters associated with it. A touch on censorship. -->
In the case that we present, we observe the times that each cell spent in both G1 and S/G2, as well as the fate observed at the end of each of those phases. As before, the fates can be operationalized as Bernoulli random variables, where a random variable of 1 for the end of G1 fate represents a transition into G2, and a 0 would represent death, and likewise, a random variable of 1 for the end of G2 fate represents division, and a 0 would represent death. Furthermore, to represent the time spent in the G1 and G2 phases, we use two gamma random variables, each parameterized by a pair of shape and scale parameters. Note that if a cell were to die in G1, no information about G2 will be known or observed. Additionally, if cell lived past a desired experiment time then, the full observation of its time spent in whatever phase it was in will also be incomplete. These are two of the many cases of censorship that affect data collection as well as how the model analyses the data, and is discussed in greater detail later on. In the previous simple case, states were simply different Bernoulli distributions with unique rate parameters. In our case, states are multivariate and are composed of two Bernoulli distributions and two gamma distributions. These states are described by six parameters in total, two Bernoulli rate parameters and two pairs of gamma shape and scale parameters. In general, any obbservable heritable phenotype that can be operationalized into a probabilistic distribution can be used to represent a state. Some examples include using a Gaussian distributed random variable to describe average cell motility, parameterized by its mean and variance. One state might describe cells that move faster while other states might describe cells that move slower. This example can be extended to include Gaussian random variables for both cell velocity and acceleration to describe cell kinematics in general.

<!-- some complementray explanation of how to work with new observations. -->
Computationally, these states are user-defined classes instantiated by the parameters that describe them. Fundamentally, if a state is to be used to analyze data collected from single-cell lineage experiments, then a state needs to include a list of parameters, a probability distribution function that takes in a tuple of observations and a list of distribution parameters and returns a likelihood, and an maximum likelihood estimator that can take in tuples of observations and return a list of parameter estimates. If one were to use their definition of a state to create computational synthetic lineages of cells, then the user would also have to provide a random variable function that takes in a list of distribution parameters of inputs, and outputs a tuple of realizations, which would be synthetic observations of cells.  

![**This figure shows the flexibility of the model and that we can use any tracktable phenotype.** ](./output/figure3.svg){#fig:3}

<!-- figure4 -->

### Model performance in the case of missing data in single-cell tracking versus fully observed dataset.
<!-- time and fate censhiprship explained. -->
A major difference between a fully observed binary tree and single-cell experimental data, is that at the end of the experiment where there is a time cut off, there are always a number of cells whom fate or phase duration is not fully or partially observed. Besides, some cells may die in the middle of the experiment and consequently, their descendants do not appear in the lineage. This effect is called right censorship where we have fully or partly unkonwn knowledge about the future. On the other hand, we may have root cells that are in the middle of their G1 or S/G2 phase when the experiment starts, and we have not observed their past. In this case, we have left-cencorship where we have missing information about the past of the experiment.  
To force the full binary tree of cells that we have created so far to be censored similar to experimental data, we have defined three conditions; 1- fate-censored, where when the cell dies we cut out its descendants, 2- time-censored, when we cut the lineage exactly at the time we intend to end the experiment, 3- both conditions applied. The results in Figure 4 are shown for a data with both types of censorship, that is the one with the highest similarity to the experimental data. The population is fed into the model, and since the true state of each cell is known, it is possible to find out how accurately the model is assigning states to the cells.

![**Performance of the model in synthetic censored and fully observed data.** (a) left column: randomly generated uncensored heterogeneous lineages of cells with 2 states; right column: randomly generated right-censored heterogeneous lineages of cells with 2 states. Cells shown in olive and salmon belong to state 0 and state 1, respectively. In the right column cells are either G1 censored (pink) or S/G2 (gold). (b) State assignment accuracy associated with the uncensored lineages. (c) State assignment accuracy associated with censored lineages. Scatter points represent the single fitting for a randomly generated population.](./output/figure4.svg){#fig:4}


### Performance on synthetic lineages

<!-- ("Figure 5") -->
<!-- explaining synthetic data. reporting some observations from the figures.-->
To indicate our model's performance in terms of state assignment accuracy and distribution parameter estimation, we created a synthetic population of cells, including a number of lineages. The synthetic data is meant to posses the properties of real experimental data where the true states of individual cells is known. To create the synthetic lineages, we assumed to have the following phenotypic  measurements: 1- whether a cell survives G1 phase and transitions to S/G2 phase, 2- whether a cell survives G2 phase and divides, 3- G1 phase duration, and 4- S/G2 phase duration. For a population with 2 known states, we need two sets of different parameters corresponding to the distributions of each phenotypic measurement. After creating a desired number of cells for each state, we randomly generated phenotypic measurements associated with the two sets of distribution parameters for the exact number of cells in each state. The observations corresponding to each state were then assigned to the cells, so that we would have a population of cells in 2 states, each cell with its observation associated with the true state it is in.    
To confirm the minimum number of cells needed in the population of interest that reach $95%$ accuracy, we created lineages with varying lengths, performed the fitting on them, and calculated the accuracy and estimated the distribution parameters. Figure 5a represents the population growth over time in which both the depth and breadth of the population increases. The state assignment accuracy is represented in Figure 5b. With $90%$ being the highest accuracy, we can conclude that $~150$ cells are enough to reach it. Figure 5c shows the difference between the predicted and true transition matrix and initial probability matrix is decreasing as the population grows larger. Figure 5d-i depict the estimated parameters (solid lines) as well as the true parameters corresponding to each phenotype. It is evident that by the time the cell number reaches $~150$, the estimation becomes fairly accurate.

![**Model performance on censored lineages of increasing breadth and depth.** Two state populations of increasing breadth (increasing number of lineages or initial cells) and of increasing depth (increasing experiment time) are analyzed. Their model performance, as based on state assignment accuracy, the error bewteen the estimated and the true transition rate matrix and the initial probability vector, and the accuracy in estimating the parameters underlying the two state distributions are shown.](./output/figure5.svg){#fig:5}



### The model performs slightly different when the states are similar or different from one another.

<!-- ("Figure 6") -->
<!-- elaboration on what we mean by different states. -->
Determining a threshold of some quality or characteristic to define "different states" in the context of cell biology is complicated. One can consider a continuous spectrum of a phenotype within a heterogeneous tumor of cells. For example, in a population one can assume infinite states corresponding to the proliferation speed. Moreover, without the assumption of discrete and finite number of states, it is hard to draw a line and separate fast from slow proliferating cells, especially if the difference is slight. To simplify this concept, we consider the cell states to be discrete and finite. In either case, if the phenotype that we are tracing as an emission is more distinct between the states, clustering cells with that phenotype and state assignment associated with cells is easier.  

<!-- How we defined parameters for different states. -->
In order to test the performance of our model given different cases as the states being similar or different, we employed Wasserstein distance metric. In our setup, the emissions include cell fate (in G1 and S/G2), G1 phase duration, and S/G2 phase duration in two states. To represent the difference between states associated with these phenotypes, we kept cell fate and S/G2 phase distribution parameters constant and varied a G1 phase distribution parameter across the states. We sweep a space where at the beginning the two states are very similar and they becomre more and more different. The shared parameters associated with the emission between the two states are ($bern_{G_1} = 0.99$, $bern_{G_2} = 0.8$, $shape_{G_1} = 12$, $shape_{G_2} = 10$, $scale_{G_2} = 5$) and the scale parameter of G1 phase for state 1 is being varied in the range of $[1, 2.5]$, while the scale parameter of G1 in state 2 is kept at $1$.  

<!-- Explaining the figure. -->
Figure 6a depicts the similarity and difference between two states in terms of cell shapes. The upper row represents state 1 with a gradual change of states over four stages, while the lower row shows state 2 which is kept the same so that the two states become more and more different over the four stages. Figure 6b presents the distribution of the cells' G1 duration for the two states and that first the two states have all their phenotypes identical, and G1 duration in state 1 changes so they become distinct. In Figure 6c, the accuracy of state assignment is shown where for the case where the two states are the same, Wasserstein distance is approximately zero and as the two states become more distinct the Wasserstein distance gets larger, meaning they are more different. Intuitively, if the states are more distinct, determining single-cells' states is more accurate. As shown in Figure 6c, the model is capable of determining the true states of single-cells when they are almost similar by almost $80%$ accuracy and reaches above $90$ when they are further distinct.

![**Model perfomance relative to the Wasserstein distance between states.** (a) A visual representation of how two states might be defined with regards to cell phenotype: as one cell evolves and changes phenotypically, the state it belongs to becomes more distant from the state the unchanging cell belongs to. This elaborates on how we can interpret having two states in cell populations where similar or different phenotypes are observed. (b) The distribution of a phenotype (namely the time spent in the G1 phase of the cell cycle) is varied in state 1 (blue) while the other state is kept constant. When cells of different states are phenotypically similar, the states themselves have probability distributions that are similar. When cells of different states are phenotypically distinct, the states are defined by distinct probability distributions. (c) The performance of our model, based on accuracy of state assignment, is shown relative to the Wasserstein distance between the two states of the cells in the populations being analyzed.](./output/figure6.svg){#fig:6}


<!-- ("Figure 7") -->
### The relative presence of states in a population is important because our model depends on non-ancillary sample statistics.

<!-- the importance of recognizing under-represented subpopulations. -->
This part explains how the model performs in the case of an under-represented population of cells.
The purpose of this section is to show the performance of the model when there are under-represented cell clusters in the population of interest. While the tools we currently have to identify heterogeneity in tumors are limited, this phenomenon indicates that it is necessary to consider underrepresented clusters in solid tumors, such as those that have gained resistance to anticancer drugs through environmental conditions, selective pressure, and cell-state transitions. In order to develop sufficient cancer therapy, it is important to recognize the existence of such tumor cells that gain resistance through non-genetic means and are not typically targeted for anticancer drugs so that we are able to accurately identify and predict the patterns that tumors gain drug resistance [@pmid:31449515]. Thus, it is relevant to consider both genetic and environmental drug resistance acquisition when determining tumor heterogeneity and its subsequent phenotypic variations in anticancer therapies.
[@doi:10.1016/j.celrep.2019.10.045].

<!-- Again, the importance of recognizing under-represented subpopulations -->
<!-- In the presence of heterogeneity with a compound that inhibits growth, at least three situations can exist: (1) The compound could have an equal effect across cell subpopulations, (2) the compound could exclusively influence growth within one subpopulation, (3) the compound could shift the relative abundance of subpopulations. Each of these situations has consequences for the potential biological mechanisms of resistance as well as how one might consider drug combinations. For example, in the second situation, one should focus efforts on drugs with the maximal effect in the unaffected subpopulation. Thus, it is crucial to be able to identify those subpopulations that form even a small proportion of the population. -->

<!-- creating the population that includes under-represented cell states, adn explaining the figure. -->
To test our model's capability to identify under-represented populations, we created a synthetic population of 2 states and varied the proportion of cells in these states while keeping the total cell number relatively constant. Figure 7a represents the successive state proportion changes from the left to the right, where the proportion of cells in the two states change in a spectrum from 0% 1 100%. The left graphical lineage in Figure 7a corresponds to a condition where the proportion of cells in state 1 and 2 are approximately 10% and 90%, respectively, with an average state assignment accuracy of 80%. Figure 7b and c depict the accuracy of state assignment while changing the proportion of states in the lineages from 0% in state 1 (equivalent to 100% in state 2) to 100% in state 1 (corresponding to 0% in state 2) for censored and fully observed populations, respectively. The difference between the performance of model in censored and uncensored population is magnified when the population is filled with more than 90% of the dominant state. Intuitively, when the population is in balance and the two state almost have the same proportion of the total cells, the accuracy of state assignment is the highest, as there are larger number of cells for each state for the model to learn. However, our model reaches the best performance when the proportion of cells in the two states are almost 80%-20%.
This implies that the model has the resolution of identifying a subpopulation 1/5 of the total cells with the highest accuracy, and even smaller proportions with approximately 80%. **This should be changed after improving the accuracy**.

![**Model performance relative to the presence of each state.** (a) A visual representation of how lineages and populations can have different proportions of their total cell population be represented by different states. (b) Populations with different ratios of cells belonging to one state over the other are made. The ratio of the number of cells in state 1 to the number of cells in state 2 increases from 1:10 to 9:10. This is achieved by synthetically creating lineages with transition rates that bias towards making one state over the other. Model performance relative to the presence of that state is shown.](./output/figure7.svg){#fig:7}


<!-- ("Figure 8") -->
### We can determine the optimal number of states to describe a population a priori using the AIC.


<!-- importance of finding the number of states in a population, then explaining how the figures were built. -->
The true number of states in a population is often unknown, and finding an estimate is rarely trivial. Determining this value is important in this context since the number of states in the population is a required parameter in our model. Akaike information criterion (AIC), a measure used for model selection, can help. AIC provides an estimate for the information lost by modeling a process. With models of varying parameters that are built using the same input data, a lower AIC score indicates better model performance. Using AIC we can determine the number of states which best fit the data, serving as an estimate for the most likely number of states in the population. To demonstrate this we first generated 8 sets of 10 simulated lineages; censored and uncensored with 1, 2, 3, and 4 true states. We then built 7 models for each dataset varying the number assumed states (1 through 7). Each of the 8 graphs (Figure 8a-h) shows the model's performance on 1 of the datasets. The number of assumed states yielding the smallest AIC represents the best prediction for the true number of states in the population. Finally, we reported the number of lineages that had the lowest AIC at each of the assumed states (1 through 7) in a histogram. This shows the number of states which best fit each lineage in the context of the model. Prediction in uncensored lineages (figure 8a-d) typically outperforms prediction in censored lineages (figure 8e-h), but this is dependent on the state distributions within the population. The smaller the number of states in the population the more accurate AIC will typically be. In Figure 8a,b,e,f with 1 and 2 true states, the number of states estimated perfectly matched with the true number of states in the population; however in figures 8d and 8h with 4 states, estimates become less definitive and less accurate. AIC performs better as a predictor without censorship and with a low number of true states. Though it is not perfect, AIC provides a reasonable estimate for the true number of states in the population.  

<!-- normalizing the AIC and explaining some specifications -->
Since AIC is a relative metric the values were normalized by subtracting the lowest AIC value for each lineage. Because of this, the best estimate will always have a normalized AIC of 0. The simulated data were generated sampling from two Bernoulli distributions to determine the fate of the cell (life/death) in G1/G2 and two Gamma distributions to determine the length of G1/G2. The parameters for the 4 states are reported in the methods. In the censored data, lineages were filtered to have more than 10 cells and more than one unique state (if there was more than 1 state given). AIC can function to estimate the true number of states in the population but functions best when data has a low number of true states and low censorship.

![**AIC** (a)-(d) AIC curves are shown for uncensored lineages made with 2 to 4 states. (e)-(h) AIC curves are shown for censored lineages made with 2 to 4 states. The number of states that yeilds the minimal AIC value represents the optimal number of states that can describe the cells in that population. Histograms are shown to count which number of states yield the minimal AIC over a number of analyzed populations.](./output/figure8.svg){#fig:8}

<!-- figure 9 -->
### The model applied on experimental data of phase-specific observations

To show the performance of the model on single-cell experimental data we employed phenotypic measurements of G1 and S/G2 phase lengths and cell fates of AU565 cells in control condition and drug-treated trials. The experiment was conducted for 96 hours and cells were imaged every 30 minutes. A fluorescent reporter translocating between the nucleus and cytoplasm was employed to indicate the phase of cells at the time of imaging. Individual cells were tracked over time manually (or using cellProfiler?). Data was imported and fitted to the model. Figure 9 depicts the most likely number of states that the model finds for different conditions of drug treatment using AIC metric. In the sample data, lineage trees with less than 5 cells were removed and for each condition we have used a population containing 9 lineage trees. Also, some simplifying assumptions were incorporated for samples treated with drugs, such as fixing the transition and initial probability matrix according to the control conditions. Moreover, from distribution analysis of the phase lengths, we realized the shape parameter of the gamma distribution remains fairly constant over different drug treatment conditions while the scale paramter changes. Along with the transition matrix and initial probability matrix, the shape parameter of the gamma distributions, associated with phase lengths were are kept constant.  

<!-- Calculating the number of parameters. -->
The AIC essentially is the likelihood of the model according to the data, penalized by a scale of the number of parameters that the model has to estimate. Calculating the number of parameters could be tricky when we have constrained set of parameters. In our case, the model is supposed to estimate a $k \times 1$ initial probability matrix, a $k \times k$ transition matrix, and a $k \times m$ matrix of state-wise parameters where $k$ is the number of states and $m$ is the number of parameters associated with observation distribution. For our phase-specific observation sets we have a total of 6 parameters including 2 bernoulli parameters, and 2 pairs of shape and scale parameters for gamma distribution. Since the row-sums for transition and initial probability matrices must be 1, these values are not independent, hence, we need to calculate the degree of freedom as follwing:  
degree of freedom = $$ k \times (k - 1) + (k - 1) + k \times (m) $$  
Given the assumption of constant shape parameters, the total number of parameters that are included in calculating the degrees of freedom becomes 6 - 2 = 4. in addition to estimating transition and initial probability matrices according to the control condition, the degree of freedom for this setting becomes $4 \times k$.

![**AIC for experimental data among 4 states of phase lengths.** (a)-(d) AIC curves for control condition and 3 concentrations of lapatinib. (e)-(h) AIC curves for control condition and 3 concentrations of gemcitabine.](./output/figure9.svg){#fig:9}
