## Results

### Our model provides more insight than population-level models into data with heterogeneous cell-types.

[//]: # (Figure 1)

As a motivating example, we present two cases of cell population growth over time, one where the population is homogeneous and all cells show consistent behaviour across their observed phenotype and one where the population can be clustered into at least two subpopulations, each with different growth rates (Figure 1b). Here we explain our motivation example based on "tumor growth rate" as the phenotype of interest. Identification of tumor growth rate has a wide range of applications in prognostics and can provide quantified therapeutic effects while different treatment modalities are in use [@doi:10.1111/j.1349-7006.1990.tb02591.x; @doi: 10.7326/0003-4819-89-1-107]. Figure 1a represents the homogeneous population over time where the cells are almost similarly dividing, and Figure 1b shows the population of cells with two distinct states shown in two different colors. Over time, the two populations look similar with regards to thier size, where the two lineages are initialized with only one cell and at the end of the experiment, both consist of 127 cells, hence, assuming the exponential growth the doubling rate, and cosequently, the growth rate for both cases will wrongly turn out to be equal. Utilizing the tHMM model we have built, we are not only able to capture the heterogeneity present in the population shown in Figure 1b, but also quantify the relative presence of these two states, and summarize the characteristics of the observed phenotypes in those states. While the growth rate (or doubling time) for both models for the homogeneous case is alike, without recognizing the heterogeneity in the population, one fails in calculating the growth rate in the second case correctly.  
As mentioned before, the capability to identify phenotypic heterogeneity in a population of cancer cells is crucial while designing anti-cancer regimens and drug combination protocols. Cell kinetics such as inter-mitotic times, cell cycle phase durations, death and division, and other phenotypic characteristics such as cell morphology, protein and transcription factor levels, etc., are important targets for cancer chemotherapy, hence, important to identify subpopulations with distinct phenotypic responses after drug exposures. Given single-cell data of any heritable phenotype, our proposed model can disentgle subpopulations that respond to the exposure differently from one another.


![TODO This figure isn't done yet.](./output/figure1.svg){#fig:1}


### How tHMM works




### States are completely user-defined and virtually any population of cells with hertiable phenotypes can be modeled.

[//]: # (Figure 2)

Because the goal of our model is to identify states, we have packaged with our model a light-weight framework for developing and using states. Cell states are defined by quantifiable differences in cell phenotype such as gene expression, post-translational modification, morphology, and production of protein [@pmid:30450444]. Additionally, cell states can be grouped by functional properties such as resistance to drugs and tumor-seeding abilities [@pmid:21854987]. Obtaining quantifiable data of cell state transitions across a lineage has relevant applications in bioengineering and drug development such as understanding and predicting the dynamic nature of cancer cells acquiring drug resistance. In systems biology, populations of cells are grouped by cell state and the data collected can be used to predict an accurate variable response in cell behavior. This cell heterogeneity, which can be caused by genetic variability as well as non-genetic mechanisms, enables cells to adapt to varying environments and is thus relevant for the study of anticancer drug development [@pmid:20371339]. We define states as multivariate probabilistic distributions, which in turn, are defined by the set of parameters that describe those very distributions. The observed phenotypes for cells in a lineage tree can be viewed as realizations of these multivariate probabilistic distributions. As a simple example, we can consider only observing the end fate of a cell. Interpreting this as a random variable, the fate of a cell can be represented as the outcome of a Bernoulli random trial: if a cell lives and divides, then the random variable is 1 and if a cell dies, then the random variable is 0. For each cell, the observed phenotype of its fate, either 1 or 0, can be seen as a sampling of the state it belongs to. The Bernoulli probability distribution is parameterized by a single rate parameter which describes the rate at which 1 is more likely than 0  as the outcome of a random trial. In this simple scenario, each state is another Bernoulli probability distribution, defined by its unique rate parameter. Ultimately, the goal of our model is to identify an optimal number of states, identify which cells belong to which states, and have the states represent the observed phenotypes of the cells in that state. Continuing the same example, if the model were to look for two states in a given cell population, then it might define the cells that divide more often into the first state, and the other set of cells that die more often into a second state. Because each state in this example is a Bernoulli probability distribution that describes the observed phenotypes of the cells that belong to that state, the first state will have a Bernoulli parameter that is larger than that of the second state. 

In the case that we present, we observe the intermitotic times of the cell spent in both G1 and G2, as well as the fate observed at the end of each of those phases. As before, the fates can be operationalized as Bernoulli random variables, where a random variable of 1 for the end of G1 fate represents a transition into G2, and a 0 would represent death, and likewise, a random variable of 1 for the end of G2 fate represents division, and a 0 would represent death. Furthermore, to represent the time spent in the G1 and G2 phases, we use two gamma random variables, each parameterized by a pair of shape and scale parameters. Note that if a cell were to die in G1, no information about G2 will be known or observed. Additionally, if cell lived past a desired experiment time then, the full observation of its time spent in whatever phase it was in will also be incomplete. These are two of the many cases of censorship that affect data collection as well as how the model analyses the data, and is discussed in greater detail later on. In the previous simple case, states were simply different Bernoulli distributions with unique rate parameters. In our case, states are multivariate and are composed of two Bernoulli distributions and two gamma distributions. These states are described by six parameters in total, two Bernoulli rate parameters and two pairs of gamma shape and scale parameters. In general, any obbservable heritable phenotype that can be operationalized into a probabilistic distribution can be used to represent a state. Some examples include using a Gaussian distributed random variable to describe average cell motility, parameterized by its mean and variance. One state might describe cells that move faster while other states might describe cells that move slower. This example can be extended to include Gaussian random variables for both cell velocity and acceleration to describe cell kinematics in general. 

Computationally, these states are user-defined classes instantiated by the parameters that describe them. Fundamentally, if a state is to be used to analyze data collected from single-cell lineage experiments, then a state needs to include a list of parameters, a probability distribution function that takes in a tuple of observations and a list of distribution parameters and returns a likelihood, and an maximum likelihood estimator that can take in tuples of observations and return a list of parameter estimates. If one were to use their definition of a state to create computational synthetic lineages of cells, then the user would also have to provide a random variable function that takes in a list of distribution parameters of inputs, and outputs a tuple of realizations, which would be synthetic observations of cells. 

![**This figure shows the flexibility of the model and that we can use any tracktable phenotype.** TODO This figure isn't done yet. Needs more work with regards to how our model can adapt to any emission.](./output/figure2.svg){#fig:2}




### Experimental data poses particular concerns when trying to identify the presence of subpopulations.

Censorship is important to consider when collecting single cell lineage data. **Figure 3** presents our model's performance on synthetic lineage data which is both uncensored and censored over populations with an increasing number of lineages. 

![**Model performance on censored and uncensored data.** Two populations with the same parameters, one uncensored, the other censored are created. (a) Representative lineages from each of the two types of populations are drwan to highlight visually the stark difference in data size that would be input to the model when considering experimental data which would be censored. In these synthetic computational lineages, censorship occurs when a cell dies (and its lineage ends) or when it lives past an experiment time. (b) The model's performance, based on state assignment accuracy, on these populations as the number of lineages in them increases is presented.](./output/figure3.svg){#fig:3}




### The model performs better with more cells.

[//]: # (Figure 4)

Here we show the performance of the model by increasing the cell numbers. The figure shows the state assignment accuracy and parameter estimations for censored lineages.

![**Model performance on censored lineages of increasing breadth and depth.** Two state populations of increasing breadth (increasing number of lineages or initial cells) and of increasing depth (increasing experiment time) are analyzed. Their model performance, as based on state assignment accuracy, the error bewteen the estimated and the true transition rate matrix and the initial probability vector, and the accuracy in estimating the parameters underlying the two state distributions are shown.](./output/figure4.svg){#fig:4}




### The Wasserstein distance represents how different states can be from one another.

[//]: # (Figure 5)

Determining a threshold of some quality or characteristic to define "different states" in the context of cell biology is complicated. One can consider a continuous spectrum of a phenotype within a heterogeneous tumor of cells. For example, in a population one can assume infinite states corresponding to the proliferation speed. Moreover, with the assumption of discrete and finite number of states, it is hard to draw a line and separate fast from slow proliferating cells, especially if the difference is slight. To simplify this concept, we consider the cell states to be discrete and finite. In either case, if the phenotype that we are tracing as an emission is more distinct between the states, clustering cells with that phenotype and state assignment associated with cells is easier.  
In order to test the performance of our model given different cases as the states being similar or different, we employed Wasserstein distance metric. In our setup, the emissions include cell fate (in G1 and S/G2), G1 phase duration, and S/G2 phase duration in two states. To represent the difference between states associated with these phenotypes, we kept cell fate and S/G2 phase distribution parameters constant and varied a G1 phase distribution parameter across the states. We sweep a space where at the beginning the two states are very similar and they becomre more and more different. The shared parameters associated with the emission between the two states are ($bern_{G_1} = 0.99$, $bern_{G_2} = 0.8$, $shape_{G_1} = 12$, $shape_{G_2} = 10$, $scale_{G_2} = 5$) and the scale parameter of G1 phase for state 1 is being varied in the range of $[1, 2.5]$, while the scale parameter of G1 in state 2 is kept at $1$.  
Figure 4a depicts the similarity and difference between two states in terms of cell shapes. The upper row represents state 1 with a gradual change of states over four stages, while the lower row shows state 2 which is kept the same so that the two states become more and more different over the four stages. Figure 4b presents the distribution of the cells' G1 duration for the two states and that first the two states have all their phenotypes identical, and G1 duration in state 1 changes so they become distinct. In Figure 4c, the accuracy of state assignment is shown where for the case where the two states are the same, Wasserstein distance is approximately zero and as the two states become more distinct the Wasserstein distance gets larger, meaning they are more different. Intuitively, if the states are more distinct, determining single-cells' states is more accurate. As shown in Figure 4c, the model is capable of determining the true states of single-cells when they are almost similar by almost $80%$ accuracy and reaches above $90$ when they are further distinct.

![**Model perfomance relative to the Wasserstein distance between states.** (a) A visual representation of how two states might be defined with regards to cell phenotype: as one cell evolves and changes phenotypically, the state it belongs to becomes more distant from the state the unchanging cell belongs to. This elaborates on how we can interpret having two states in cell populations where similar or different phenotypes are observed. (b) The distribution of a phenotype (namely the time spent in the G1 phase of the cell cycle) is varied in state 1 (blue) while the other state is kept constant. When cells of different states are phenotypically similar, the states themselves have probability distributions that are similar. When cells of different states are phenotypically distinct, the states are defined by distinct probability distributions. (c) The performance of our model, based on accuracy of state assignment, is shown relative to the Wasserstein distance between the two states of the cells in the populations being analyzed.](./output/figure5.svg){#fig:5}




### The relative presence of states in a population is important because our model depends on non-ancillary sample statistics.

[//]: # (Figure 6)

This parts explains how the model performs in the case of an under-represented population of cells.
The purpose of this section is to show the performance of the model when there are under-represented cell clusters in the population of interest. 
While the tools we currently have to identify heterogeneity in tumors are limited, this phenomenon indicates that it is necessary to consider underrepresented clusters in solid tumors, such as those that have gained resistance to anticancer drugs through environmental conditions, selective pressure, and cell-state transitions. In order to develop sufficient cancer therapy, it is important to recognize the existence of such tumor cells that gain resistance through non-genetic means and are not typically targeted for anticancer drugs so that we are able to accurately identify and predict the patterns that tumors gain drug resistance [@pmid:31449515]. Thus, it is relevant to consider both genetic and environmental drug resistance acquisition when determining tumor heterogeneity and its subsequent phenotypic variations in anticancer therapies. 
[@doi:10.1016/j.celrep.2019.10.045].
In the presence of heterogeneity with a compound that inhibits growth, at least three situations can exist: (1) The compound could have an equal effect across cell subpopulations, (2) the compound could exclusively influence growth within one subpopulation, (3) the compound could shift the relative abundance of subpopulations. Each of these situations has consequences for the potential biological mechanisms of resistance as well as how one might consider drug combinations. For example, in the second situation, one should focus efforts on drugs with the maximal effect in the unaffected subpopulation. Thus, it is crucial to be able to identify those subpopulations that form even a small proportion of the population. To test our model's capability to identify under-represented populations, we created a synthetic population of 2 states and varied the proportion of cells in these states while keeping the total cell number relatively constant. Figure 6a represents the successive state proportion changes from the left to the right, where the proportion of cells in state 1 (and 2) changes from 10% (90%) to 90% (10%). Figure 6b depicts the accuracy of state assignment while changing the proportion of states in the lineages. The left graphical lineage in Figure 6a corresponds to a condition where the proportion of cells in state 1 and 2 are approximately 10% and 90%, respectively, with and average state assignment accuracy of 80%. Intuitively, when the population is in balance and the two state almost have the same proportion of the total cells, the accuracy of state assignment is the highest, as there are larger number of cells for each state for the model to learn. However, our model reaches the best performance when the proportion of cells in the two states are almost 80%-20%.
This implies that the model has the resolution of identifying a subpopulation 1/5 of the total cells with the highest accuracy, and even smaller proportions with approximately 80%.

![**Model performance relative to the presence of each state.** (a) A visual representation of how lineages and populations can have different proportions of their total cell population be represented by different states. (b) Populations with different ratios of cells belonging to one state over the other are made. The ratio of the number of cells in state 1 to the number of cells in state 2 increases from 1:9 to 9:1. This is achieved by synthetically creating lineages with transition rates that bias towards making one state over the other. Model performance relative to the presence of that state is shown.](./output/figure6.svg){#fig:6}




### We can determine the optimal number of states to describe a population a priori using the AIC.

[//]: # (Figure 7)

In this part we explain the AIC figures.

![**AIC.** AIC curves are shown for (a)-(d) uncensored and (e)-(h) censored lineages made with 2 to 4 states. The number of states that yields the minimal AIC value represents the optimal number of states that can describe the cells in that population. Histograms are shown to count which number of states yield the minimal AIC over a number of analyzed populations.](./output/figure7.svg){#fig:7}


