## Discussion

Big picture: Importance of this study:

1. Phenotypic intratumoral heterogeneity induces resistance to drug exposure, so we need to account for them in designing anticancer therapy.
2. It is important to take heritability and cell-cell relations into account while studying the evolution and state transitions.
3. Previous studies have investigated identifying phenotypic heterogeneity using models.
4. The tHMM can help with uncovering the underlying heterogeneity using single-cell lineage data using only experimental obervations.

Intratumoral heterogeneity in cancer cells enables them to adjust to therapy, and advancement to more malignant stages. For instance, cells with variable growth rates have different susceptibility to estressors such as drug exposure [@doi:10.1016/j.gde.2011.10.001]. Even in the absence of genetic mutations, non-genetic stable variants that are heritable could serve as substrates for natural selection [@doi:10.1038/nrg2556]. Therefore, it is vital to identify the sources of non-genetic heterogeneity and provide strategies that include the effect of these variabilities in new therapy regiments. Non-genetic phenotypes such as cell cycle durations have been shown to be highly correlated within and across cell generations through lineage analysis [@doi:10.1101/373258]. Previous studies have investigated the dynamic of cell state transitions based on the molecular states, differential gene expressions, or cell types but mostly assumed the number and identity of the states are granted [@doi:10.1093/bioinformatics/btp456; @doi:10.1016/j.cell.2011.07.026; @doi:10.1038/s41598-019-46926-x; @doi:10.1016/j.cels.2016.10.015]. Using a state-space model, Nakashima et al., introduced a modified expectation-maximization algorithm to infer cell states from mother-daughter relations and division times [@doi:10.1093/bioinformatics/btaa040], however, they also assumed there are fast- and slow-growing states within the population. Identifying the number of latent states and statistically characterizing such states, while accounting for cell-cell relations has not been the focus of related studies so far. In this paper, we introduced a tree-based hidden Markov model that classifies single cells of a heterogeneous population based on solely their over-time phenotypic traits into merely homogeneous subpopulations. We implemented a modified version of Baum-Welch algorithm for G1 and S/G2 cell cycle phase durations and cell fate of tracked single cells in a lineage structure that could be trained with virtually any number of phenotypic measurements. As an application, single cell lineage tracking data of AU565 cancer cells treated with a range of lapatinib and gemcitabine concentrations was inputted to the model. Utilizing the BIC metric and the EM algorithm, the model predicted 6 and 5 distinct subpopulations for lapatinib and gemcitabine-treated data, respectively, and provided statistical characteristics of each state.

Additional findings and how this fits to existing literature:

1. tHMM shows high sensitivity for under-represented sub-populations.

2. tHMM has proved to be more accurate than K-means clustering which shows the importance of accounting for history of cells and their relations.

3. tHMM properly handles time-censorship using survivoral function of distributions, which is barely the case in the literature.

Insightful findings from lapatinib treatment fitting, and comparing to some literature about effects of lapatinib on breast cancer cells:
- State with higher possibility of cell death, have less cell cycle phase extension (state 3)
- All states have G1 extension, some more and some less.
- Comparison...

Insightful findings from gemcitabine treatment fitting and comparing to some literature about effects of gemcitabine on breast cancer cells:
- The effects appear only when the concentration of gemcitabine reaches 10 nM or higher. The drug effects are drastic across concentrations and less smooth compared to lapatinib.
- State 5 with high inward probability, has a very high chance of cell death.
- Comparison...

Critical analysis of our findings and limitations of the study:

1. Discritizing the state, while in reality they are coninuous variables, is a challenging assumption which may cause inaccuracy to some extent.

2. tHMM cannot perform with high accuracy when the states are too close.

3. In the current setting, the measurements are assumed to be independent. If observations are not independent variables, we should account for that in the likelihood function.

4. The experiment time is limited, and we may not have captured slow-fluctuations. Also, it is hard to collect this type of data for longer times due to experimental limitations.

5. Without further experiments, the molecular interpretations of each state is not clear. 

future directions:

1. Sequencing cells that have been assigned states after fitting and revealing what could be the distinguishing factor for separating those cells assigned to different states.

2. Integerating information about heterogeneity due to mutations to have a comprehensive insight.

3. Based on the findings from sequencing cells, design combination experiments and assess the efficacy.

4. The model is applicable to a wide range of lineage-tree data to unravel the underlying heterogeneity.
