## Discussion

<!-- Big picture: Importance of this study -->
Intratumoral heterogeneity in cancer cells enables them to adjust to therapy, and advancement to more malignant stages. For instance, cells with variable growth rates have different susceptibility to estressors such as drug exposure [@doi:10.1016/j.gde.2011.10.001]. Even in the absence of genetic mutations, non-genetic stable variants that are heritable could serve as substrates for natural selection [@doi:10.1038/nrg2556]. Therefore, it is vital to identify the sources of non-genetic heterogeneity and provide strategies that include the effect of these variabilities in new therapy regiments. In this paper, we introduced a tree-based hidden Markov model that classifies single cells of a heterogeneous population based on solely their over-time phenotypic traits into merely homogeneous subpopulations. We implemented a modified version of Baum-Welch algorithm for phenotypic observations of tracked single cells in a lineage structure that could be trained with virtually any number of phenotypic measurements. As an application, single cell lineage tracking data of AU565 cancer cells treated with a range of lapatinib and gemcitabine concentrations was inputted to the model. The data included G1 and S/G2 cell cycle phase durations and cell fate at either of these phases. Utilizing the BIC metric, the model predicted 6 and 5 distinct subpopulations for lapatinib and gemcitabine-treated data, respectively, and provided statistical characteristics of each state.

<!-- additional findings and how this fits to existing literature -->
It is important to account for the history and relations between cells when studying cell-to-cell heterogeneity. Non-genetic phenotypes such as cell cycle durations have been shown to be highly correlated within and across cell generations through lineage analysis [@doi:10.1101/373258]. The dynamic of cell state transition has been determined using lineage trees and single-cell end point measurements, where cell states are identified by differential gene expression levels [@doi:10.1016/j.cels.2016.10.015]. Nakashima et al., [@doi:10.1093/bioinformatics/btaa040] using a state-space model introduced a modified expectation-maximization algorithm, accompanied by a corrections for survivorship bias for a lineage tree data to model phenotypic variability in bacteria. However, identifying the number of latent states and statistically characterizing such states has not been done so far.
The model showed relatively high sensitivity to the populations with under-represented sub-populations and proved efficacious with more distinct state features. Particularly in comparison to K-means clustering algorithm, the average accuracy of state assignment was higher by $15%$. The model was evaluated at an experimentally feasible data size and classified all types of synthetic populations of 2 true states with the accuracy higher than 95% for a minimum of ~100 cells in the population. The tHMM showed relatively high sensitivity to the populations with under-represented subpopulations and proved efficacious with more distinct state features (Figs. @fig:wass1, @fig:wass2). Particularly in comparison to K-means clustering algorithm, the average accuracy of state assignment was higher by 15% (Fig. @fig:wass).Another advancement in the current work, is using suvival functions to properly handle the missing information in the form of time censorship both at the beginning (left-truncated) and the end of the experiment (right-censored) (Fig. @fig:censor).
<!-- critical analysis of our findings and limitations of the study -->
In reality, cells live in a continuum of phenotypic and behavioral state [@doi:10.1038/s41591-021-01233-9] and it is more or less arbitrary where to draw the line that discritizes these states. On the other hand, building a model with continuous states has many challenges. That is why we should anticipate errors in some degree in clustering cells with tHMM. Also, the model is dependent on the distance between the existing states and may perform poorly when the states are close or when cells possessed high variance in parameter estimation (Fig. @fig:wass). Moreover, in our setting, the cell cycle phase durations and fates are assumed to be independent variables. If the measurements are not independent, then we need to adjust for that in the likelihood function.

Because our experimental data goes as along as 96 hours it only covers up to 5 or 6 generations of cells. The fact that whether cells after several generations reach a stable equilibrium is not determinable for our dataset. Further, some traits such as resistance may develop over more generations and not detectable in short-term experiments. On the other hand, data collection for more than 96 hours becomes challenging due to various environmental factors, among which phototoxicity and cell stress [@pmid:27213923]. The model operates equally well whether the population of interest owns a pre-existing phenotypic heterogeneity (synthetic data figures) or the cells acquire diverse phenotypes as a result of drug treatment (experimental analysis), but it cannot recognize which is the case. The model selection validated by the BIC metric asserted the reliability of tHMM in predicting the most likely number of states within a population (Fig. @fig:sBIC).
<!-- future directions -->
As a follow up study, We intend to design assays to experimentally validate the number of states predicted by tHMM, and study the extent and sources of differences between the identified states. The current version of this pipeline is most accurate when populations consist of at least 100 cells. Transitions from therapy-susceptible to therapy-resistant states are well detected by the model, which will prove useful for identifying mutant subtypes in tumors and leading to more optimal therapies for cancer treatment. The tHMM may further be used for drug screening as a single cell, rather than population-based, means of quantifying the potency of novel therapies in eliminating all subpopulations within a tumor. The pipeline will provide researchers and clinicians with an improved classification of heterogeneity among cells, or any other tree-structured data, and provide information about latent changes in cellular identity.
This model although insightful, might be incomplete to shed light into all the sources that lead to heterogeneity. Upgrading the model or integrating it with other studies such that it could also take genetic sources of variations into account, would be even more useful.
This model is applicable not only for single-cell lineage data, but also to any other type of lineage data to cluster the individual elements based on their measurements.
